components/director/internal/destinationfetchersvc/handler.go

```diff
@@ -47,6 +50,18 @@ func (h *handler) SyncTenantDestinations(writer http.ResponseWriter, request *ht
 		return
 	}
 
+	isTenantSubscribed, err := h.validateTenantSubscription(ctx, tenantID)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("Failed to validate tenant %q subscription", tenantID)
+		http.Error(writer, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if !isTenantSubscribed {
+		log.C(ctx).Infof("Tenant %q is not subscribed", tenantID)
+		http.Error(writer, fmt.Sprintf("Tenant %q is not subscribed", tenantID), http.StatusBadRequest)
```

Comment:
Is "bad request" the most accurate status? Usually, it means a client error, something wrong with the client http request. I'm not saying "don't use error 400 here" but rather questioning if it is suitable.  
P.S. Same applies for the above L56 and for the other handler function

Reply:
It is not the most accurate status but the whole handler is implemented like that - if there is a problem, return 400

Reply:
I'm cool with 400s if they are being widely used in the service.